### PR TITLE
[ArrayManager] ENH: ArrayManager.convert without block fallback

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
xref https://github.com/pandas-dev/pandas/issues/39146

### CODE REVIEW

File: pandas/core/internals/array_manager.py [hunk 1]
```diff
@@ -515,13 +516,19 @@ def convert(
...
         timedelta: bool = True,
     ) -> ArrayManager:
-        return self.apply_with_block(
-            "convert",
-            copy=copy,
-            datetime=datetime,
-            numeric=numeric,
-            timedelta=timedelta,
-        )
+        def _convert(arr):
```
Comment (jreback) [inline pandas/core/internals/array_manager.py:519]: not pretty but ok

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/internals/array_manager.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/core/internals/array_manager.py:519]: What's not pretty about it? The inline function?

Comment (jreback) [reply pandas/core/internals/array_manager.py:519]: soft_convert_objects can just do this directly right? (or the copy keyword is not functioning)

Comment (jorisvandenbossche) [reply pandas/core/internals/array_manager.py:519]: Currently `soft_convert_objects` assumes the input is object dtype, and doesn't do such checking. I assume the check could be moved into ``soft_convert_objects`` as well (although eg for Block, which has an ObjectBlock, it knows it is ndarray[object], and such an additional check would be unnecessary)

Comment (jbrockmendel) [reply pandas/core/internals/array_manager.py:519]: BTW soft_convert_objects is only used for ObjectBlock.convert ATM, so no real harm in making it handle non-object with these extra two lines (though to get effective simplification in the BM code might need to tweak copy kwarg behavior)

Comment (jreback) [reply pandas/core/internals/array_manager.py:519]: ok if would revisit / consolidate logic at some point here.

**Review by jreback — APPROVED** [review]

**Pull request merged**